tests/test_web_api.py

```diff
@@ -0,0 +1,307 @@
+from hashlib import md5
+from mock import patch
+from os import path as ospath
+from regparser.web.jobs.models import job_status_values
+from regparser.web.jobs.utils import (
+    eregs_site_api_url,
+    file_url,
+    status_url
+)
+from regparser.web.jobs.views import FileUploadView as PatchedFileUploadView
+from rest_framework.test import APITestCase
+from tempfile import NamedTemporaryFile
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+
+fake_pipeline_id = "f726e1e0-d43c-4eb7-9274-681ddaba427a"
+fake_email_id = "4774f0f6-b53e-4b34-821e-fc8ae5e113fe"
+
+
+def fake_add_redis_data_to_job_data(job_data):
+    return job_data
+
+
+def fake_delete_eregs_job(job_id):
+    return None
+
+
+def fake_redis_job(args, timeout=60*30, result_ttl=-1):
+    return type("FakeRedisJob", (object, ), {"id": fake_pipeline_id})
+
+
+def fake_queue_email(job, statusurl, email_address):
+    return fake_email_id
+
+
+# Even though these functions are in the ``utils`` module, the code paths we're
+# following mean we're encountering them in the ``views`` namespace:
+@patch("regparser.web.jobs.views.add_redis_data_to_job_data",
+       fake_add_redis_data_to_job_data)
+@patch("regparser.web.jobs.views.delete_eregs_job", fake_delete_eregs_job)
+@patch("regparser.web.jobs.views.queue_eregs_job", fake_redis_job)
+@patch("regparser.web.jobs.views.queue_notification_email", fake_queue_email)
+class PipelineJobTestCase(APITestCase):
+
+    defaults = {
+        "clear_cache": False,
+        "destination": eregs_site_api_url,
+        "use_uploaded_metadata": None,
+        "use_uploaded_regulation": None,
+        "regulation_url": "",
+        "status": "received"
+    }
+
+    def _postjson(self, data):
+        return self.client.post("/rp/job/pipeline/", data, format="json")
+
+    def _stock_response_check(self, expected, actual):
+        """
+        Since we're using a lot of fake values, the tests for them will always
+        be the same.
+        """
+        for key in expected:
+            self.assertEqual(expected[key], actual[key])
+        self.assertIn(actual["status"], job_status_values)
+
+    def test_create_ints(self):
+        data = {
+            "cfr_title": 0,
+            "cfr_part": 0,
+            "notification_email": "test@example.com"
+        }
+        response = self._postjson(data)
+
+        expected = {k: data[k] for k in data}
+        expected.update(self.defaults)
```

Comment:
It doesn't matter in this scenario as there's no overlap, but shouldn't we begin with the defaults and `update` them? Also, you can use `dict(my_dict)` for a quick, shallow copy